packages/account/src/providers/transaction-request/helpers.ts

```diff
@@ -93,3 +95,61 @@ export const cacheRequestInputsResourcesFromOwner = (
       messages: [],
     } as Required<ExcludeResourcesOption>
   );
+
+/**
+ * @hidden
+ *
+ * Get the number of burnable assets in the transaction request.
+ *
+ * @param opts - The transaction request to get the burnable asset count from.
+ * @returns The number of burnable assets in the transaction request.
+ */
+export const getBurnableAssetCount = (
+  baseAssetId: string,
+  opts: Pick<TransactionRequest, 'inputs' | 'outputs'>
+) => {
+  const { inputs, outputs } = opts;
+  const coinInputs = new Set(inputs.filter(isRequestInputCoin).map((input) => input.assetId));
+  // If there is a message input without data, we need to add the base asset to the set
+  if (inputs.some(isRequestInputMessageWithoutData)) {
```

Comment:
```suggestion
  if (inputs.some((i) => isRequestInputMessage(i) && bn(i.amount).gt(0))) {
```

Even though messages with data (non-coin messages ) cannot be used to pay for the transaction fee, they are spendable and they can have amounts, which will result in a UTXO being created by a `ChangeOutput` or/and `CoinOutput`

Reply:
I was not aware, thanks for the lowdown.

Reply:
62dc663ebc3c93dddfdd92030dad5e029947e05d